test/authrestapi_test.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/yunusemreayhan/goAuthMicroService/models"
+	"github.com/yunusemreayhan/goAuthMicroService/restapi"
+)
+
+const (
+	testUsername  = "test3"
+	testPassword  = "test3"
+	testPassword2 = "test3_2"
+	testEmail     = "test3@gmail.com"
+)
+
+var client = &http.Client{
+	Transport: &http.Transport{
+		IdleConnTimeout: 30 * time.Second,
+	},
+	Timeout: 30 * time.Second,
+}
+
+var server *httptest.Server
+
+var httpHandler *http.Handler
+
+func bodyCloser(body io.ReadCloser) {
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Default().Printf("Body.Close error : [%v]\n", err)
+		}
+	}(body)
+}
+func TestAuthApi(t *testing.T) {
+	if httpHandler == nil {
+		handler, err := restapi.GetAPIHandler()
+		if err != nil {
+			t.Fatal("get api handler", err)
+		}
+		httpHandler = &handler
+	}
+	if server == nil {
+		server = httptest.NewServer(*httpHandler)
+	}
```

Comment:
It would be better to construct handler and server before tests. In the 'init()' function for example.